crds/operators.coreos.com_catalogsources.yaml

```diff
@@ -86,10 +86,18 @@ spec:
                   mediatype:
                     type: string
               image:
-                description: Image is an operator-registry container image to instantiate
-                  a registry-server with. Only used when SourceType = SourceTypeGrpc.
+                description: Image is a container image of an index which hosts a
+                  collection of operator bundles. Only used when SourceType = SourceTypeGrpc.
                   If present, the address field is ignored.
                 type: string
+              imageSecret:
```

Comment:
I don't think this disambiguates secret usage as much as treating access to a catalog and its contents as a whole would; i.e. giving access to a catalog requires giving access to pull its image and all images that it references. In this world, we have one set of secrets referenced by the CatalogSource object, which are automatically bound to the default serviceaccount of the catalog pods as well as copied and automatically bound to operator serviceaccounts. Having two fields makes for a more complex UX. I also don't really see the case for segregating access between an index and its content, at that point why not just have separate catalogs for private content. We also played around with this concept with `CatalogSourceConfigs`, which we have since abandoned.

Further, I posit that it may have been an oversight that secrets weren't bound to the catalogsource pod's serviceaccount  to begin with.

Reply:
>I don't think this disambiguates secret usage as much as treating access to a catalog and its contents as a whole would; i.e. giving access to a catalog requires giving access to pull its image and all images that it references.

>I also don't really see the case for segregating access between an index and its content, at that point why not just have separate catalogs for private content.

These are assuming that the index author has permission to give access to every bundle image s/he is including in the index. 
Consider this scenario: I am a private index author, who wants to include an operator with a bundle image that's public, and another operator who's bundle image requires a pull secret. 
Bottom line: Separation of concern - index image is one thing, and bundle images are another, and it's probably presumptuous to require users to give access to everything with just one pull secret to rule them all.  

 > In this world, we have one set of secrets referenced by the CatalogSource object, which are automatically bound to the default serviceaccount of the catalog pods as well as copied and automatically bound to operator serviceaccounts.

Quoting @exdx as my response: 

"@nhale’s suggestion to bind the secret to the default SA in the catalog pod’s namespace also makes sense, although it makes the assumption that OLM should always use the default SA  to pull down catalog images. I believe the default SA is always present in a namespace but one can imagine it can be heavily restricted/disabled. We don’t provide a way for a user to specify which SA should be used in the catalog pod."
 
> Having two fields makes for a more complex UX

I am of the opinion that `spec.secrets` might have made sense in the previous world,  but in this world where secrets could be for a variety of categories of images, i.e index image or bundle image etc, `spec.secrets` just doesn't convey anything and should be renamed too. But we could start with 
```
spec:
       image: bar/foo/image
       imageSecret: 
                 - secretName: baz
```

> I am also being cautious because each new field added is something that we need to support for quite some time.

Could you expand on how the support for this might get complex?

Reply:
> I don't think this disambiguates secret usage as much as treating access to a catalog and its contents as a whole would; i.e. giving access to a catalog requires giving access to pull its image and all images that it references.
> 

This is very true, we should probably treat the access to the catalog itself and its content as the same. In that case the SA approach makes more sense.